.github/workflows/java-model-diff.yml

```diff
@@ -0,0 +1,98 @@
+name: Models as Data
+
+on:
+  push:
+    branches:
+      - main
+  workflow_dispatch:
+    inputs:
+      projects:
+        description: "The projects to generate models for"
+        required: true
+        default: '["netty/netty"]'
+  pull_request:
+    branches:
+      - main
+    paths:
+      - "java/ql/src/utils/model-generator/**/*.*"
+
+jobs:
+  model-diff:
```

Comment:
Give this job a human-readable `name` too. It can be the same as the workflow name.